<launch>
  <arg name="launch_driver" default="false" />

  <arg name="pcap"  default=""/>
  <arg name="device_ip" default="192.168.1.201"/>
  <arg name="lidar_port"  default="2368"/>
  <arg name="gps_port"  default="10110"/>
  <arg name="scan_phase"  default="0.0"/>
  <arg name="return_mode" default="Dual"/>
  <arg name="dual_return_distance_threshold" default="0.1"/>
  <arg name="model" default="Pandar64"/>
  <arg name="frame_id" default="pandar"/>
  <arg name="calibration"  default="$(find-pkg-share pandar_pointcloud)/config/64.csv"/>
  <arg name="manager" default="pandar_nodelet_manager"/>

<group>
  <push-ros-namespace namespace="/sensing/lidar/top"/>


  <!-- pandar cloud -->
  <node pkg="pandar_pointcloud" exec="pandar_cloud_node" name="pandar_cloud" output="screen" >
    <remap from="pandar_points" to="pointcloud_raw" />
    <remap from="pandar_points_ex" to="pointcloud_raw_ex" />
    <param name="scan_phase" value="$(var scan_phase)"/>
    <param name="return_mode" value="$(var return_mode)"/>
    <param name="dual_return_distance_threshold"  value="$(var dual_return_distance_threshold)"/>
    <param name="model" value="$(var model)"/>
    <param name="frame_id" value="$(var frame_id)"/>
    <param name="device_ip" value="$(var device_ip)"/>
    <param name="calibration" value="$(var calibration)"/>
  </node>
</group>
</launch>
